packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1242,7 +1242,7 @@ class RenderEditable extends RenderBox {
   @override
   void performLayout() {
     _layoutText(constraints.maxWidth);
-    _caretPrototype = Rect.fromLTWH(0.0, _kCaretHeightOffset, cursorWidth, preferredLineHeight - 2.0 * _kCaretHeightOffset);
+    _caretPrototype = Rect.fromLTWH(0.0, kCaretHeightOffset, cursorWidth, preferredLineHeight - 2.0 * kCaretHeightOffset);
```

Comment:
The caret is made to have some offset so that it is slightly smaller than the full line height.